Makefile

```diff
@@ -7,7 +7,7 @@ LICENSE     := MIT
 
 # Build variables
 BUILD_DIR   := bin
-VERSION     := $(shell git describe --tags --exact-match 2>/dev/null || git describe --tags 2>/dev/null)
+VERSION     := $(shell git describe --always --tags --dirty 2>/dev/null || echo "0.0.0" )
```

Comment:
Why default to "0.0.0"?

Reply:
Should it default to the last commit hash instead?

Reply:
Defaulting to `"0.0.0"` doesn't apply to us here, but that's a generic (copy/paste from another Makefile of mine) to ensure build never fails. `VERSION` will be as follow:

- `vx.y.z`: we're exactly on a tag `vx.y.z`
- `vx.y.z-a-<hash>`: we're `a` commits ahead of tag `vx.y.z` and the current commit is `<hash>`
- `<hash>`: we're on commit `<hash>` and no previous tag found (i.e. the very beginning of development)

The main reason for this change was to have a way to do `make patch`, `make minor` and `make major`